src/m365/pp/commands/solution/solution-publish.spec.ts

```diff
@@ -0,0 +1,254 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../../appInsights';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+import * as PpSolutionGetCommand from './solution-get';
+const command: Command = require('./solution-publish');
+
+describe(commands.SOLUTION_PUBLISH, () => {
+  let commandInfo: CommandInfo;
+  //#region Mocked Responses
+  const validEnvironment = '4be50206-9576-4237-8b17-38d8aadfaa36';
+  const validId = '00000001-0000-0000-0001-00000000009b';
+  const validName = 'Solution name';
+  const envUrl = "https://contoso-dev.api.crm4.dynamics.com";
+  const validParameterXmlData = {
+    ParameterXml: '<importexportxml><canvasapps><canvasapp>new_test_eb178</canvasapp></canvasapps><entities><entity>new_test</entity></entities></importexportxml>'
+  };
+  const validSolutionComponentsResult = {
+    value: [
+      {
+        'msdyn_componentlogicalname': 'canvasapp',
+        'msdyn_name': 'new_test_eb178'
+      },
+      {
+        'msdyn_componentlogicalname': 'entity',
+        'msdyn_name': 'new_test'
+      }
+    ]
+  };
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
```

Comment:
Hi, just popping in to notify you that we should update this line to:

```
sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
```

to align it with our latest change: #4215